x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/plan/logical/command/sys/SysTypesTests.java

```diff
@@ -5,107 +5,148 @@
+        "INTERVAL_HOUR_TO_MINUTE", "INTERVAL_HOUR_TO_SECOND", "INTERVAL_MINUTE_TO_SECOND",
+        "GEO_POINT", "GEO_SHAPE", "SHAPE", "UNSUPPORTED", "NESTED", "OBJECT");
 
-        List<String> names = asList("BYTE", "LONG", "BINARY", "NULL", "INTEGER", "SHORT", "HALF_FLOAT", "FLOAT", "DOUBLE", "SCALED_FLOAT",
-                "IP", "KEYWORD", "TEXT", "BOOLEAN", "DATE", "TIME", "DATETIME",
-                "INTERVAL_YEAR", "INTERVAL_MONTH", "INTERVAL_DAY", "INTERVAL_HOUR", "INTERVAL_MINUTE", "INTERVAL_SECOND",
-                "INTERVAL_YEAR_TO_MONTH", "INTERVAL_DAY_TO_HOUR", "INTERVAL_DAY_TO_MINUTE", "INTERVAL_DAY_TO_SECOND",
-                "INTERVAL_HOUR_TO_MINUTE", "INTERVAL_HOUR_TO_SECOND", "INTERVAL_MINUTE_TO_SECOND",
-                "GEO_POINT", "GEO_SHAPE", "SHAPE", "UNSUPPORTED", "NESTED", "OBJECT");
+    public void testSysTypes() {
+        Tuple<Command, SqlSession> cmd = sql("SYS TYPES");
 
-        cmd.execute(session(), wrap(p -> {
+        cmd.v1().execute(cmd.v2(), wrap(p -> {
             SchemaRowSet r = (SchemaRowSet) p.rowSet();
             assertEquals(19, r.columnCount());
             assertEquals(SqlDataTypes.types().size(), r.size());
             assertFalse(r.schema().types().contains(DataTypes.NULL));
-            // test numeric as signed
+            // test first numeric (i.e. BYTE) as signed
             assertFalse(r.column(9, Boolean.class));
             // make sure precision is returned as boolean (not int)
             assertFalse(r.column(10, Boolean.class));
             // no auto-increment
             assertFalse(r.column(11, Boolean.class));
 
             for (int i = 0; i < r.size(); i++) {
-                assertEquals(names.get(i), r.column(0));
+                assertEquals(ALL_TYPES.get(i), r.column(0));
                 r.advanceRow();
             }
 
         }, ex -> fail(ex.getMessage())));
     }
 
+    public void testUnsignedLongFiltering() {
+        Set<SqlVersion> versions = new HashSet<>(List.of(
+            SqlVersion.fromId(INTRODUCING_UNSIGNED_LONG.id - SqlVersion.MINOR_MULTIPLIER),
+            INTRODUCING_UNSIGNED_LONG,
+            SqlVersion.fromId(INTRODUCING_UNSIGNED_LONG.id + SqlVersion.MINOR_MULTIPLIER),
+            SqlVersion.fromId(Version.CURRENT.id)
+        ));
+        versions.add(null);
+        for (SqlVersion version : versions) {
+            for (Mode mode : Mode.values()) {
+                Tuple<Command, SqlSession> cmd = sql("SYS TYPES", mode, version);
+
+                cmd.v1().execute(cmd.v2(), wrap(p -> {
+                    SchemaRowSet r = (SchemaRowSet) p.rowSet();
+                    List<String> types = new ArrayList<>();
+                    r.forEachRow(rv -> types.add((String) rv.column(0)));
+                    List<String> expectedTypes = ALL_TYPES.stream()
+                        .filter(t -> Mode.isDriver(mode) == false || isTypeSupportedInVersion(DataTypes.fromTypeName(t), version))
```

Comment:
Another way of doing this is to:
1. have dedicated lists for base types +unsigned and check with and without.
2. check its presence only in versions w/ and w/o it.

Reply:
Thanks, refactored.